File: wifi4eu-common/src/main/java/wifi4eu/wifi4eu/common/dto/model/PublicationCallDTO.java
package wifi4eu.wifi4eu.common.dto.model;

import javax.validation.constraints.NotNull;
import java.util.Date;

public class PublicationCallDTO {

    private Long callId;
    @NotNull
    private String url;
    @NotNull
    private Date startDate;
    @NotNull
    private Date endDate;

    public PublicationCallDTO(){}

    public PublicationCallDTO(Long callId, String url, Date startDate, Date endDate) {
        this.callId = callId;
        this.url = url;
        this.startDate = startDate;
        this.endDate = endDate;
    }

    public Long getCallId() {
        return callId;
    }

    public void setCallId(Long callId) {
        this.callId = callId;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Date getStartDate() {
        return startDate;
    }

    public void setStartDate(Date startDate) {
        this.startDate = startDate;
    }

    public Date getEndDate() {
        return endDate;
    }

    public void setEndDate(Date endDate) {
        this.endDate = endDate;
    }
}

File: wifi4eu-web/src/main/angular/src/app/shared/swagger/model/PublicationCallDTO.ts
import * as models from './models';
import {Type} from "class-transformer";

export interface PublicationCallDTO {
    callId?: number;

    url: string;

    startDate: Date;

    endDate: Date;

}




/*default implementation one might extend from (or use as is) */


export class PublicationCallDTOBase  implements PublicationCallDTO{

    callId?:  number ;


    url:  string ;

@Type(() => Date)
    startDate:  Date ;

@Type(() => Date)
    endDate:  Date ;

}
